src/main/golo/assertions.golo

```diff
@@ -0,0 +1,293 @@
+    }
+  })
+}
+
+----
+Helper to test predicate
+
+- *param* `predicate` this is a closure (FunctionReference)
+- *param* `allStackTrace` if true, the complete Java StackTrace is displayed, if false, only Golo stuff is displayed (Boolean)
+- *param* `getSuccessMessage` this is a closure that return a string (FunctionReference)
+- *param* `onSuccess` this is the callBack if success (FunctionReference)
+- *param* `getErrorMessage` this is a closure that return a string (FunctionReference)
+- *param* `exitWhenFailed` if true, then the program aborts at the first error (Boolean)
+- *param* `onError` this is the callBack if error (FunctionReference)
+----
+function assert = |predicate, allStackTrace, getSuccessMessage, onSuccess, getErrorMessage, exitWhenFailed, onError| {
+
+  if(not_windows) { reset() }
+  try {
+    # run the predicate
+    let result = predicate()
+    # if predicate is false then display error message
+    # and throw exception
+    if(result isnt true) {
+      if(not_windows) { fg_red() }
+      let errorMessage = getErrorMessage()
+      println(errorMessage)
+      report: failed(report: failed()+1)
+      throw Exception(errorMessage)
+    } else {
+      # predicate is verified, then display success message
+      # and execute success callback if exists
+      if(not_windows) { fg_green() }
+      println(getSuccessMessage())
+      if(not_windows) { reset() }
+      report: passed(report: passed()+1)
+      Option(onSuccess): either(|callBack| -> callBack(result), {})
+    }
+  } catch(err) {
+      if(allStackTrace) { # if allStackTrace is true, display all Java stacktrace
+        err: getStackTrace(): asList(): each(|stackTraceLine| {
+          # if current line of stacktrace is about golo, display details and line that generates the error
+          if(
+            stackTraceLine: toString(): contains(".golo:")
+            and not stackTraceLine: toString(): startsWith("gololang.Assertions")
+          ) {
+              if(not_windows) { fg_red() }
+              println("  " + stackTraceLine)
+              getDetailsOfGoloError(stackTraceLine)
+
+          } else { # display "Java stuff"
+              if(not_windows) { fg_black() }
+              println("  " + stackTraceLine)
+          }
+        })
+      } else { # if allStackTrace is false, display only "Golo stuff"
+        err: getStackTrace(): asList(): filter(|stackTraceLine| ->
+          stackTraceLine: toString(): contains(".golo:")
+            and not stackTraceLine: toString(): startsWith("gololang.Assertions")
+        ): each(|stackTraceLine| {
+          println("  " + stackTraceLine)
+          getDetailsOfGoloError(stackTraceLine)
+          if(not_windows) { fg_red() }
+        })
+      }
+      if(not_windows) { reset() }
+
+      # if exitWhenFailed is true, then exit of tests (program) at first error and display report
+      if(exitWhenFailed) {
+        displayTestsReport()
+        java.lang.System.exit(1)
```

Comment:
Why don't you just throw an exception? Then you could feed the results to any processing function (not just necessarily print them systematically). This is where it could integrate nicely with a test harness module, see @danielpetisme's ideas.